src/types/table.ts

```diff
@@ -0,0 +1,41 @@
+export interface TableContextProps {
+  expandedRows: string[]
+  setExpandedRows: (rowKey: string) => void
+  collapseRows: () => void
+}
+
+export interface TableProps {
+  children: React.ReactNode | React.ReactNode[]
```

Comment:
Let's try to deconstruct these elements, for example:

```
import { ReactNode } from 'react'

(...)
children: ReactNode
(...)
```

This probably should also work with `Fragment`